app/src/main/java/org/simple/clinic/sync/LastSyncedState.kt

```diff
@@ -13,6 +13,15 @@ data class LastSyncedState(
     val lastSyncSucceededAt: Instant? = null
 ) {
 
+  fun withProgress(syncProgress: SyncProgress): LastSyncedState {
+    check(syncProgress != SUCCESS) { "Use success() instead if the sync is successful!" }
```

Comment:
This is a bit confusing api.
Can we instead have all three progress states with different apis?

Reply:
Good point, I'll do this.